File: ts/project/.sdk/tm/csharp/core/Result.cs
// ProjectName SDK - operation result.

using Voxgig.Struct;

namespace ProjectNameSdk;

public class Result
{
    public bool Ok;
    public int Status = -1;
    public string StatusText = "";
    public Dictionary<string, object?> Headers = new();
    public object? Body;
    public Exception? Err;
    public object? Resdata;
    public Dictionary<string, object?>? Resmatch;

    // Feature extensions: pagination signals (paging feature) and the
    // incremental item iterator (streaming feature).
    public Dictionary<string, object?>? Paging;
    public bool Streaming;
    public Func<IEnumerable<object?>>? Stream;

    public Result(Dictionary<string, object?>? resmap)
    {
        resmap ??= new Dictionary<string, object?>();

        if (StructUtils.GetProp(resmap, "ok") is bool b)
        {
            Ok = b;
        }

        var s = StructUtils.GetProp(resmap, "status");
        if (s != null)
        {
            Status = Helpers.ToInt(s);
        }

        if (StructUtils.GetProp(resmap, "statusText") is string st)
        {
            StatusText = st;
        }

        if (StructUtils.GetProp(resmap, "headers") is Dictionary<string, object?> hm)
        {
            Headers = hm;
        }

        Body = StructUtils.GetProp(resmap, "body");

        if (StructUtils.GetProp(resmap, "err") is Exception er)
        {
            Err = er;
        }

        Resdata = StructUtils.GetProp(resmap, "resdata");

        if (StructUtils.GetProp(resmap, "resmatch") is Dictionary<string, object?> rmm)
        {
            Resmatch = rmm;
        }
    }
}

File: ts/project/.sdk/tm/kotlin/core/Result.kt
package KOTLINPACKAGE.core

import java.util.function.Supplier

import KOTLINPACKAGE.utility.struct.Struct

/** The processed outcome of one operation. */
@Suppress("UNCHECKED_CAST")
class Result(resmap: Map<String, Any?>?) {

  var ok: Boolean = false
  var status: Int = -1
  var statusText: String = ""
  var headers: MutableMap<String, Any?> = linkedMapOf()
  var body: Any? = null
  var err: RuntimeException? = null
  var resdata: Any? = null
  var resmatch: MutableMap<String, Any?>? = null

  // Feature extensions: pagination signals (paging feature) and the
  // incremental item iterator (streaming feature).
  var paging: MutableMap<String, Any?>? = null
  var streaming: Boolean = false
  var stream: Supplier<Iterator<Any?>>? = null

  init {
    val o = Struct.getprop(resmap, "ok")
    if (o is Boolean) {
      this.ok = o
    }

    val s = Struct.getprop(resmap, "status")
    if (s is Number) {
      this.status = Helpers.toInt(s)
    }

    val st = Struct.getprop(resmap, "statusText")
    if (st is String) {
      this.statusText = st
    }

    val h = Struct.getprop(resmap, "headers")
    if (h is MutableMap<*, *>) {
      this.headers = h as MutableMap<String, Any?>
    }

    this.body = Struct.getprop(resmap, "body", null)

    val e = Struct.getprop(resmap, "err")
    if (e is RuntimeException) {
      this.err = e
    }

    this.resdata = Struct.getprop(resmap, "resdata", null)

    val rm = Struct.getprop(resmap, "resmatch")
    if (rm is MutableMap<*, *>) {
      this.resmatch = rm as MutableMap<String, Any?>
    }
  }
}
